### DISCUSSION TITLE
Pengurusan CMS public web

### DISCUSSION THREAD

**Discussion by rohaizanr** [Dev4w4n/e-masjid.my#130] [Ideas]
Perlu tukar ke wordpress? security concerns?
Bina CMS sendiri public web.

![Screenshot 2024-03-10 at 6 17 32 PM](https://github.com/Dev4w4n/e-masjid.my/assets/11247162/4610c308-5630-4c84-9983-0d8fbfc49433)

**Comment by m-amirazmi** [comment]
Rasanya boleh up penggunaan NextJS utk ni. Dan kita build custom CMS untuk ni.
Utk public-web punya design, saya ada jumpa beberapa design yang menarik di Dribbble, tapi 1 ni menarik perhatian saya. Maybe saya cuba buat POC NextJS app locally dulu.

[https://dribbble.com/shots/16665610-Taqwa-Islamic-Center-Masjid-Website](https://dribbble.com/shots/16665610-Taqwa-Islamic-Center-Masjid-Website)

![image](https://github.com/Dev4w4n/e-masjid.my/assets/60350614/7c2d5852-d35d-4920-a2c8-f2b5c5b2f4e7)

**Reply by rohaizanr** [reply]
Saya harap boleh buat sekali dengan peti cadangan sedia ada.

Soalan, custom CMS ni kena ada API dan DB untuk support?

**Reply by m-amirazmi** [reply]
Haah mmg akan ada API yg maybe nk kena pull utk display rough analytics mcm jumlah kutipan jumaat, nama masjid. N maybe jugak kena ada db storage utk store data2 mcm siapa imam, siapa bilal, siapa bg tazkirah utk sekian2 hari. Sy ingt mcm nk simpan as JSON file kt frontend je sbb bnda tu guna kt public-web je kn.

**Reply by rohaizanr** [reply]
module ini saya boleh kata mcm standalone module yang takda tight dependencies dengan module yang lain.

future boleh add-on, tapi initial design kena future proof.

**Reply by m-amirazmi** [reply]
Haah, thats the plan.

**Comment by dinayub** [comment]
Menarik kalau buat cms ala2 divi, breakdance, framer, mcm web builder

**Reply by m-amirazmi** [reply]
Kalau web builder, baik pakai wordpress je terus. Nk buat custom giler gak tu. Tp klau web builder style yg di perlukan, better guna wordpress je

**Reply by m-amirazmi** [reply]
And, sy xde experience or x prnh sentuh wordpress hahaha

**Comment by m-amirazmi** [comment]
[Sanity IO](https://www.sanity.io/)

Untuk CMS Nextjs, kalau tak kisah sangat pasal component building and component dtg dr developer dan fix, rasanya boleh pakai ni.

Ni boleh allow user utk manage content, gambar n video

**Reply by m-amirazmi** [reply]
[Payload CMS](https://payloadcms.com/blog/the-ultimate-guide-to-using-nextjs-with-payload)

Ni pun ok gak. Nanti saya try tgk. Nk buat custom boleh, tapi saya risau saya ambik masa lama sgt.

**Comment by rohaizanr** [comment]
Soalan: Yang ni untuk edit main page public-web sahaja atau boleh create custom sub pages

![Screenshot 2024-03-26 at 3 10 39 PM](https://github.com/Dev4w4n/e-masjid.my/assets/11247162/810d04d4-2d45-4aaa-bf0e-f1cf78ab937e)

**Reply by rohaizanr** [reply]
@m-amirazmi  ?
